OpenSim/Simulation/MarkersReference.cpp

```diff
@@ -77,19 +77,13 @@ void MarkersReference::populateFromMarkerData(const MarkerData& aMarkerData)
     _markerNames.assign(nm, "");
     _weights.assign(nm, get_default_weight());
 
-    int index = 0;
-    // Build flat lists (arrays) of marker names and weights in the same order as the marker data
-    for(int i=0; i<tempNames.getSize(); i++){
+    for (int i = 0; i < tempNames.getSize(); i++) {
         const std::string &name = tempNames[i];
         _markerNames[i] = name;
```

Comment:
The decision to track an individual marker reference is left to the `InverseKinematicsSolver` (`setupGoals()`), which operates on a `Model` and uses the common subset of markers (between the Model and Reference). But the `MarkersReference` itself has no knowledge of the model.